src/implementations/vanilla/components/mega-menu/mega-menu.js

```diff
@@ -1504,6 +1506,12 @@ export class MegaMenu {
     };
     // If the focus trap is active, deactivate it
     this.focusTrap.deactivate();
+    // Focus on the open button in mobile or on the formerly expanded item in desktop.
+    if (!this.isDesktop && this.open) {
+      this.open.focus();
```

Comment:
this causes the hamburger menu to be always the focus element when you click anywhere on the page or select any button